### PR TITLE
bugfixes to getPeaklist and getReducedPeaklist

### PR DESCRIPTION
**Pull request by korseby**
Hi Steffen,

this PR fixes some minor issues in getPeaklist and getReducedPeaklist when cleanup is set to TRUE.

Best wishes, *Kristian

### CODE REVIEW

**Comment by sneumann** [comment]
Seems to not work yet:
```
> peaklist.reduced <- getReducedPeaklist(an)
Error in apply(X = pl.tab, MARGIN = 2, FUN = function(x) { : 
  dim(X) must have a positive length
Calls: getReducedPeaklist ... eval -> eval -> tapply -> lapply -> FUN -> FUN -> app
```

**Comment by korseby** [comment]
Hi @sneumann ,

The automatic test failed because the getPeaklist(object)  of object@xcmsSet was only done on 1 sample... This has been corrected in my branch.

Please note that my commit also incorporates the bugfix to issue 22 https://github.com/sneumann/CAMERA/issues/22 ;)

Best wishes, *Kristian

File: R/xsAnnotate.R [hunk 1]
```diff
@@ -1362,67 +1362,71 @@ setGeneric("getReducedPeaklist", function(object, method = "median", intval = "i
...
+    subset_samp <- c((ncol(peaklist)-nrow(object@xcmsSet@phenoData)-1) : (ncol(peaklist)-1-1))
+  }
+  
   # Create reduced peak list
   peaklist_reduced <- by(data=peaklist, INDICES=peaklist$pcgroup, FUN=function(plist) {
     if (nrow(plist) > 1) {
       # Subset intensities
-      pl.tab <- plist[, subset_samp]
-            
+      pl.tab <- as.data.frame(plist[, subset_samp])
```
Comment (sneumann) [inline R/xsAnnotate.R]: Would you get the same result with `plist[, subset_samp, drop=FALSE]` ?

**Comment by korseby** [comment]
Yes, it returns the same result when using `plist[, subset_samp, drop=FALSE]`. I changed that and tested that. Thanks.

**Pull request merged**